test/include/autoinstall/AutoinstallAskInclude_test.rb

```diff
@@ -0,0 +1,292 @@
+#!/usr/bin/env rspec
+
+root_path = File.expand_path('../..', __FILE__)
+ENV["Y2DIR"] = File.join(root_path, 'src')
+
+require "yast"
+require_relative "../../../src/include/autoinstall/ask.rb"
+require_relative "../../../src/modules/Profile"
+
+describe "Yast::AutoinstallAskInclude" do
+  module DummyYast
+    class AutoinstallAskClient < Yast::Client
+      def main
+        Yast.include self, "autoinstall/ask.rb"
+      end
+
+      def initialize
+        main
+      end
+    end
+  end
+
+  let(:client) { DummyYast::AutoinstallAskClient.new }
+  let(:base_ask) do
+    { "type" => "string", "question" => "hostname?", "default" => "my.site.de",
+      "help" => "Some help", "dialog" => 0, "element" => 0,
+      "stage" => "initial" }
+  end
+  let(:profile) { { "general" => { "ask-list" => ask_list } } }
+  let(:result) { :ok }
```

Comment:
I use this kind of approach when I want to redefine some value in a nested context. For example, this time the `result` (rename to `button_pressed` following your suggestion) it's used in the before block at top-level and I just want to redefine the button at nested levels. Redefining the let just at the start of the context, make it clear enough for me.

Having said that, I don't have an strong position on that anyway.